main_test.go

```diff
@@ -10,11 +10,45 @@ import (
 func TestTimeGranularity(t *testing.T) {
 	baseTime := time.Date(2017, 7, 11, 12, 0, 0, 0, time.UTC)
 
-	start, end := startEndFromGranularity(baseTime, "day")
+	start, end := startEndFromGranularity(baseTime, "day", false)
 	assert.Equal(t, start, time.Date(2017, 7, 11, 0, 0, 0, 0, time.UTC))
 	assert.Equal(t, end, time.Date(2017, 7, 12, 0, 0, 0, 0, time.UTC))
 
-	start, end = startEndFromGranularity(baseTime, "week")
+	start, end = startEndFromGranularity(baseTime, "hour", false)
+	assert.Equal(t, start, time.Date(2017, 7, 11, 12, 0, 0, 0, time.UTC))
+	assert.Equal(t, end, time.Date(2017, 7, 11, 13, 0, 0, 0, time.UTC))
+
+	// Simulate timestamps that cross timezones in PT vs UTC
+	baseTime = time.Date(2017, 7, 11, 4, 0, 0, 0, time.UTC)
+
+	start, end = startEndFromGranularity(baseTime, "day", false)
+	assert.Equal(t, start, time.Date(2017, 7, 11, 0, 0, 0, 0, time.UTC))
+	assert.Equal(t, end, time.Date(2017, 7, 12, 0, 0, 0, 0, time.UTC))
+
+	start, end = startEndFromGranularity(baseTime, "day", true)
 	assert.Equal(t, start, time.Date(2017, 7, 10, 0, 0, 0, 0, time.UTC))
```

Comment:
Oh wait a minute. I think I remember why I didn't touch `startEndFromGranularity` now - the timestamps coming in are PT, but the output is also in PT. See here https://github.com/Clever/s3-to-redshift/blob/master/main.go#L166 AND https://github.com/Clever/s3-to-redshift/blob/master/redshift/redshift.go#L453-L454:

So it doesn't really matter what timezone we are operating under since the comparisons will be made in the same Redshift column anyways.

Does that sound about right? If so I'll remove these changes to `startEndFromGranularity`